jujugui/static/gui/src/app/utils/analytics.js

```diff
@@ -0,0 +1,87 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2017 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('analytics', function(Y) {
+
+  let ns = Y.namespace('juju');
+
+  /**
+    Send analytics to GA.
+
+    @param {String} category A sensible category name. Generally the
+      component name is a good start.
+    @param {String} action Some identifiable action.
+    @param {String} label Name the event. This might be enough for
+      some events.
+    @param {Object} value A single depth object of extra information.
+  */
+  ns.sendAnalytics = function(controllerAPI, dataLayer) {
```

Comment:
s/sendAnalytics/analyticsSender/?

so that then we can have `sendAnalytics = analyticsSender(...)` and it's slightly clearer that this is a factory function?
This is a take it or leave it suggestion, as you already changed this API so much, do this only if you like it.